pkg/catalog/routes.go

```diff
@@ -353,3 +349,21 @@ func getDefaultWeightedClusterForService(meshService service.MeshService) servic
 		Weight:      constants.ClusterWeightAcceptAll,
 	}
 }
+
+// listTrafficTargetPermutations creates a list of TrafficTargets for each source and destination pair.
+func listTrafficTargetPermutations(name string, srcServiceList []service.MeshService, destServiceList []service.MeshService) []trafficpolicy.TrafficTarget {
+	trafficPolicies := make([]trafficpolicy.TrafficTarget, 0, len(srcServiceList)*len(destServiceList))
+
+	for _, destService := range destServiceList {
+		for _, srcService := range srcServiceList {
+			trafficTarget := trafficpolicy.TrafficTarget{
+				Name:        name,
```

Comment:
Wondering if using the same name for the `trafficpolicy.Traffictarget` objects is a good idea. Is this intentional? 
I can see this as a means to group different `trafficpolicy.TrafficTarget` objects to SMI's TrafficTarget policy, but not clear what's the benefit in doing so. Should we instead rename this to `source -> destination` similar to https://github.com/openservicemesh/osm/blob/b52d4c382b501307d3de11a42a72df0b8d34d800/pkg/catalog/routes.go#L343

Reply:
I think naming it as `fmt.Sprintf("%s->%s", srcMeshSvc, dstMeshSvc)` makes sense in the allow all scenario since we are building our own traffic policies. But in this scenario the user/customer is applying/naming a specific policy so when we want to debug this name would help us identify which policy its coming from. I  think its better to keep the name of `trafficpolicy.Traffictarget`

Reply:
I think it would be easier to debug from the logs if the logs were in the form of `source -> dest`. When multiple objects match with the same name, requires extra effort to debug.

Reply:
I see your point @shashankram ,  how about `trafficpolicy.Traffictarget:source -> dest`  ? Cause a given `src-> dest` can come from multiple policies, adding the name will make it more explicit

Reply:
At the end of the day that Name is only relevant for debugging, and not used anywhere else, so augmenting it as <TrafficTarget-name: source -> dest> should be okay for the SMI case.

Reply:
@ksubrmnn please could you update the name to `trafficpolicy.Traffictarget:source -> dest`

Reply:
@ksubrmnn we didn't want to completely revert the naming, but to change it to what's described above, just to aid debugging.

Reply:
I made the change and then saw this comment. Fixing now.

Reply:
updated